### PR TITLE
recommendation workflow

### PR DESCRIPTION
**Pull request by kevingreer**
This includes a change to the apiary documentation. Instead of having a recommendation count in the user model, there will be one in the episode model. The client only needs the latter.
#64 
------------------------------
At first episode 1 has no recommendations
screen shot 2017-09-06 at 11 14 37 pm

Now we'll create a recommendation for it
screen shot 2017-09-06 at 11 22 25 pm

And the episode is updated accordingly
screen shot 2017-09-06 at 11 24 16 pm

After creating a recommendation for episode 2 we can get both that I've recommended
screen shot 2017-09-06 at 11 27 49 pm

And get the recommendations for episode 1
screen shot 2017-09-06 at 11 28 41 pm

Lastly we can delete a recommendation
screen shot 2017-09-06 at 11 29 02 pm

And the episode recommendation count for episode 2 goes back down to 0
screen shot 2017-09-06 at 11 39 30 pm

### CODE REVIEW

**Comment by ChenJesse** [comment]
LGTM!

**Review by ChenJesse — APPROVED** [review]

File: py/src/app/pcasts/models/recommendation.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  __bind_key__ = 'db'
+  __table_args__ = (
+      UniqueConstraint('user_id', 'episode_id'),
+  )
+
+  id = db.Column(db.Integer, primary_key=True)
+  user_id = db.Column(db.Integer, db.ForeignKey('users.id', ondelete='CASCADE'))
+  episode_id = db.Column(db.Integer, nullable=False)
+
+  user = db.relationship('User', cascade='all,delete')
```
Comment (ChenJesse) [inline py/src/app/pcasts/models/recommendation.py:15]: Not sure if the `cascade='all,delete'` is correct here
Haven't tested it with `recommendation` or `bookmark`, but I realized with `subscription` and `following` that when i delete a subcription/following, it deletes the associated user as well. When I delete the `cascade='all,delete'`, the problem is fixed. Maybe the cascade is going in the opposite direction from what we wanted?

Comment (Jma353) [reply py/src/app/pcasts/models/recommendation.py:15]: Yes it is going in the opposite direction - we should only have `cascade='all,delete'` for relationships that define related model that should be deleted if the current model is deleted.

Comment (Jma353) [reply py/src/app/pcasts/models/recommendation.py:15]: I think from working quickly we might have a couple of these issues - we should have a PR to fix those.  Merge this and I'll make that PR.

**Pull request merged**